File: P104 Maximum Depth of Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int maxDepth(TreeNode root) {
         if(root == null)    
             return 0;
             
         // Non-recursive, use level order triversal
         ArrayList<TreeNode> q = new ArrayList<TreeNode>();
         q.add(root);
         int depth = 0;
         
         while(!q.isEmpty()) {
             ArrayList<TreeNode> next = new ArrayList<TreeNode>();
             for(TreeNode node : q) {
                 if(node.left != null)   next.add(node.left);
                 if(node.right != null)  next.add(node.right);
             }
             q = new ArrayList<TreeNode>(next);
             depth++;
         }
         
         return depth;
    }
}

File: P104 Maximum Depth of Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        """ 
        recursive:
            # @param root, a tree node
            # @return an integer
        if root == None:
            return 0
        return max(self.maxDepth(root.left),self.maxDepth(root.right))+1
        """
        if root == None:
            return 0
        nodeStack = [root];
        depthStack = [1];
        maxDepth = 0;
        while len(nodeStack)>0:
            node = nodeStack.pop();
            depth = depthStack.pop();
            maxDepth = maxDepth if maxDepth > depth else depth
            if node.left != None:
                nodeStack.append(node.left)
                depthStack.append(depth+1)
            if node.right != None:
                nodeStack.append(node.right)
                depthStack.append(depth+1)
        return maxDepth